src/components/BookCard/BookCard.tsx

```diff
@@ -1,13 +1,43 @@
+import { useDispatch } from "react-redux";
+import type { AppDispatch } from "../../store/store";
 import type { Book } from '../../domain/books/types'
+import { addToCart } from "../../domain/cart/cartSlice";
+import "./BookCard.css";
 
 /* 
   We would define the structure of the props in the same file as the component.
   Since, this component has the same props as Type Book, we would assign it to it props.
 */
+
 type BookProps = Book;
 
-export const BookCard = (props: BookProps): JSX.Element => {
+export const BookCard = (book: BookProps): JSX.Element => {
+
+  const dispatch: AppDispatch = useDispatch();
```

Comment:
You can also use AppDispatch as an argument to useDispatch
`const dispatch = useDispatch<AppDispatch>()`